Lib/test/test_inspect/test_inspect.py

```diff
@@ -4623,6 +4623,106 @@ class D2(D1):
         self.assertEqual(inspect.signature(D2), inspect.signature(D1))
 
 
+class TestSignatureFromFrame(unittest.TestCase):
+    def test_from_frame(self):
+        ns = {}
+        def inner(a, /, b, *e, c: int = 3, d, **f) -> None:
+            ns['fr'] = inspect.currentframe()
+
+        inner(1, 2, d=4)
+        self.assertEqual(str(inspect.Signature.from_frame(ns['fr'])),
+                         '(a=1, /, b=2, *e, c=3, d=4, **f)')
```

Comment:
The "default" values are really misleading. I think it would be better not to populate default values at all. If users want to know the value of a local variable, they can inspect the frame themselves IMHO.

_Or_ you could add an optional argument governing this behavior, defaulting to False.

Reply:
If we don't use frame locals to populate default parameter values, I don't think that adding an option for this behavior is worth it.

Reply:
I think we can add an option to turn defaults off, with adding them by defaults.